tests/python/integration/test_model_compile.py

```diff
@@ -0,0 +1,151 @@
+# pylint: disable=missing-docstring
+import concurrent.futures as cf
+import os
+import shlex
+import subprocess
+import sys
+import tempfile
+from itertools import product
+
+import tvm
+
+from mlc_chat.model import MODEL_PRESETS
+from mlc_chat.support.constants import MLC_TEMP_DIR
+
+OPT_LEVEL = "O2"
+DEVICE2TARGET = {
+    "cuda": tvm.target.Target(
+        {
+            "kind": "cuda",
+            "arch": "sm_86",
+            "max_threads_per_block": 1024,
+            "max_num_threads": 1024,
+            "max_shared_memory_per_block": 49152,
+            "thread_warp_size": 32,
+        }
+    ),
+    "rocm": tvm.target.Target(
+        {
+            "kind": "rocm",
+            "mtriple": "amdgcn-amd-amdhsa-hcc",
+            "mcpu": "gfx1100",
+            "thread_warp_size": 32,
+            "max_threads_per_block": 1024,
+            "max_num_threads": 256,
+            "max_shared_memory_per_block": 65536,
+        }
+    ),
+    "vulkan": tvm.target.Target(
+        {
+            "kind": "vulkan",
+            "max_threads_per_block": 1024,
+            "max_num_threads": 256,
+            "max_shared_memory_per_block": 32768,
+            "thread_warp_size": 1,
+            "supports_int16": 1,
+            "supports_float32": 1,
+            "supports_int32": 1,
+            "supports_int8": 1,
+            "supports_16bit_buffer": 1,
+            "supports_float16": 1,
+        }
+    ),
+}
+DEVICE2SUFFIX = {
+    "cuda": "so",
+    "rocm": "so",
+    "vulkan": "so",
+}
+MODELS = list(MODEL_PRESETS.keys())
+QUANTS = [  # TODO(@junrushao): use `list(mlc_chat.quantization.QUANTIZATION.keys())`
```

Comment:
per discussion with @MasterJH5574: we will enable `q3f16_0` and `q4f16_0` once related bugs are fixed

Reply:
These two are supposed to be fixed by this PR https://github.com/apache/tvm/pull/16429. Let's enable after it gets merged.